app/controllers/OnlineTestController.scala

```diff
@@ -97,11 +91,32 @@ trait OnlineTestController extends BaseController {
   def resetOnlineTests(appId: String) = Action.async(parse.json) { implicit request =>
     withJsonBody[ResetOnlineTest] { resetOnlineTest =>
       appRepository.getOnlineTestApplication(appId).flatMap {
-        case Some(onlineTestApp) => onlineTestingService.resetTests(onlineTestApp, resetOnlineTest.tests,
+        case Some(onlineTestApp) => phase1TestService.resetTests(onlineTestApp, resetOnlineTest.tests,
           resetOnlineTest.actionTriggeredBy).map ( _ => Ok )
         case _ => Future.successful(NotFound)
       }
     }
+  }
+
+  def resetPhase2OnlineTest(appId: String) = Action.async(parse.json) { implicit request =>
+    withJsonBody[ResetPhase2Test] { resetPhase2Test =>
+
+      def reset(onlineTestApp: OnlineTestApplication, actionTriggeredBy: String) =
+        phase2TestService.resetTests(onlineTestApp, actionTriggeredBy)
+          .map { _ =>
+            Ok(Json.toJson(ResetPhase2TestStatus(success = true)))
+          }
+          .recover {
+            case _: ResetLimitExceededException =>
+              Ok(Json.toJson(ResetPhase2TestStatus(success = false, resetLimitExceeded = true)))
+            case _: CannotResetPhase2Tests =>
+              Ok(Json.toJson(ResetPhase2TestStatus(success = false)))
```

Comment:
Perhaps this should be a 404, given that the CannotResetPhase2Tests exception inherits from NotFound. Certainly returning a 200 is not very 'RESTFul'. I would argue that ResetLimitExceededException could be HTTP code 424 and CannotResetPhase2Tests a 404. That way you don't need to return a success flag. What do you think?